explorer/interpreter/resolve_names.cpp

```diff
@@ -273,15 +273,30 @@ auto NameResolver::ResolveNames(Expression& expression,
       break;
     }
     case ExpressionKind::SimpleMemberAccessExpression: {
+      // If the left-hand side of the `.` is a namespace or alias to namespace,
+      // resolve the name.
       auto& access = cast<SimpleMemberAccessExpression>(expression);
       CARBON_ASSIGN_OR_RETURN(std::optional<ValueNodeView> scope,
                               ResolveNames(access.object(), enclosing_scope));
+      if (!scope) {
+        break;
+      }
 
-      // If the left-hand side of the `.` is a namespace, resolve the name.
-      // TODO: Look through aliases.
-      if (scope && isa<NamespaceDeclaration>(scope->base())) {
-        auto ns_it =
-            namespace_scopes_.find(&cast<NamespaceDeclaration>(scope->base()));
+      Nonnull<const AstNode*> base = &scope->base();
+      // recursively resolve aliases
+      auto namespace_alias =
+          [](const AstNode& base) -> std::optional<Nonnull<const AstNode*>> {
+        if (const auto* alias = dyn_cast<AliasDeclaration>(&base)) {
+          return alias->resolved_declaration();
+        }
+        return std::nullopt;
+      };
+      while (auto new_base = namespace_alias(*base)) {
+        base = *new_base;
+      }
```

Comment:
```suggestion
      // Recursively resolve aliases.
      while (const auto* alias = dyn_cast<AliasDeclaration>(base); alias && alias->resolved_declaration()) {
        base = *alias->resolved_declaration();
      }
```

Reply:
this was also my first approach. But looks like c++ doesn't support while with initializer. 
closest is for without update, but in that case control flow is incorrect because init is run only once not every iteration.

Reply:
What am I thinking:

for (const auto* alias = dyn_cast<AliasDeclaration>(base); alias && alias->resolved_declaration(); ) {